### DISCUSSION TITLE
argo-workflows/webhooks/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#10458] [General]
# argo-workflows/webhooks/

https://argoproj.github.io/argo-workflows/webhooks/

**Comment by OneCricketeer** [comment]
> type of webhook the account can be used for, e.g. `github`.

And what other types are allowed...? 

> Many clients can send events

Oh? I've resorted to looking at the source code, seems like this only accepts a very **limited** number ([GitHub, GitLab, and BitBucket](https://github.com/argoproj/argo-workflows/tree/master/server/auth/webhook)). 

How do we open this API up to allow simple HTTP requests like `curl` to start a workflow, since it seems like it is [not currently possible](https://github.com/argoproj/argo-workflows/blob/v2.11.4/server/auth/webhook/interceptor.go#L78-L94)?

**Reply by OneCricketeer** [reply]
Answering own question. The only requirement for using this API from outside of the options listed above, is by passing in any a non-empty `Authorization` header to the [Events API](https://argoproj.github.io/argo-workflows/events/). The value is not validated beyond its length (see `interceptor.go`).

**Comment by ShyunnY** [comment]
What should I do if I want to trigger the workflow running through github webhook? Are there any corresponding examples?

**Reply by OneCricketeer** [reply]
Use Argo Events Github source, combined with an Argo workflow trigger.
https://argoproj.github.io/argo-events/eventsources/setup/github/

**Reply by agilgur5** [reply]
The linked Workflows [Events](https://argoproj.github.io/argo-workflows/events/) docs has examples of how a webhook requested can be formatted, but not a GitHub-specific example. Feel free to add such an example to the docs!
